lib/extendedkey.ex

```diff
@@ -65,22 +82,24 @@ defmodule Bitcoinex.ExtendedKey do
       [num | tick] = String.split(level, ["'", "h"])
       nnum = String.to_integer(num)
 
-      if nnum >= @softcap or nnum < 0 do
+      if nnum >= @min_hardened_child_num or nnum < 0 do
         raise(ArgumentError, message: "invalid derivation path")
```

Comment:
This is an error on my part. it should be @max_hardened_child_num.

Reply:
This current function does not allow numbers greater than 2^31, assuming all hardened child_nums will use ' or h. Is this assumption safe?

Reply:
I think its worth keeping this as is

Reply:
fixed and refactor a bit. Please let me know your thought